### DISCUSSION TITLE
PT1000 Sensor @ ESP8266EX Smart Socket

### DISCUSSION THREAD

**Discussion by der-seemann** [arendst/Tasmota#19715] [Q&A]
Hello Tasmota Community,

I hope you're all doing well. I'm writing to seek some assistance and guidance regarding my project to integrate a PT1000 temperature sensor with an ESP8266EX-powered smart socket running Tasmota.

**Project Overview:**
I have a PT1000 temperature sensor (fixed in place, can't change it for now) that I'd like to read directly using the free ADC (Analog-to-Digital Converter) pin on my ESP8266EX-based smart socket. My setup involves connecting the PT1000 sensor between VCC (3.3 V), a 10k resistor, and ground, with the ADC pin connected between the PT1000 and the 10k resistor.

**The Issue:**
I've configured Tasmota to use the ADC pin as a temperature sensor input. However, I'm encountering two issues:
1. The temperature reading is stuck at approximately 16°C, which doesn't seem accurate.
2. When I set to ADC input, it consistently shows a reading of 1024.

**My Questions:**
1. What could be causing the temperature reading to be stuck at around 16°C? Is there something I might be missing in my setup or configuration?
2. Why does the ADC input mode consistently show a reading of 1024? Is there an issue with my wiring or configuration?

I would greatly appreciate any insights, advice, or troubleshooting steps that you can offer to help me resolve this issue and successfully integrate my PT1000 sensor with Tasmota on my ESP8266EX smart socket.

Thank you in advance for your assistance, and I look forward to your valuable input.

Best regards,
Dave

**Comment by barbudor** [comment]
What kind of socket is it?
Beware that on socket with power measurement, the ESP is directly connected to the mains without any isolation. If this is the case, do not attempt to connect a sensor as there is a high risk of electric shock or even death.

Sincerely, considering the cost of a small  ESP based board, I wouldn't loose time in tinkering a socket

**Reply by sfromis** [reply]
And getting analog input value of 1024 implies very low resistance, what I see when directly connected to 3.3V. Likely bad wiring.

I do agree with barbudor about sockets not being good devices to add stuff to.

**Comment by lalo-uy** [comment]
PT1000 sensores have very little variation anda need a preamplifier before
the adc. You will go better with a dedícate interface like max31865 that is
suportted bye tasmota.

**Comment by Noschvie** [comment]
[#14420 : How to use MAX31865 and PT1000 ](https://github.com/arendst/Tasmota/discussions/14420)
